From 885640eba864a67b2ad091573c84b3d14cd2c2ad Mon Sep 17 00:00:00 2001
From: Nicolin Chen <b42378@freescale.com>
Date: Fri, 29 Nov 2013 17:44:39 +0800
Subject: [PATCH 0675/1543] ENGR00290229 ASoC: fsl: Drop snd_soc_dapm_sync()
 in imx-wm8962

commit 1b7581c93f613d74329bebfcfe3ff1470a6a2ffd from
git://git.freescale.com/imx/linux-2.6-imx.git

As DAPM would do the sync() for us, we don't need to handle it by ourselves.
And leaving snd_soc_dapm_sync() here is dangerous because it would disable
the clock from WM8962 during the short period of the output route changing
since we don't leave the alternative route's enanbling to this machine driver
but to DAPM core.

Acked-by: Wang Shengjiu <b02247@freescale.com>
Signed-off-by: Nicolin Chen <b42378@freescale.com>
(cherry picked from commit c4dcde0b5119262cbb75c5136422af2071bffbda)
---
 sound/soc/fsl/imx-wm8962.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8962.c b/sound/soc/fsl/imx-wm8962.c
index 6e9d3d3..aae7238 100644
--- a/sound/soc/fsl/imx-wm8962.c
+++ b/sound/soc/fsl/imx-wm8962.c
@@ -108,7 +108,6 @@ static int hpjack_status_check(void)
 		snd_soc_dapm_enable_pin(&priv->codec->dapm, "Ext Spk");
 		ret = 0;
 	}
-	snd_soc_dapm_sync(&priv->codec->dapm);
 
 	envp[0] = "NAME=headphone";
 	envp[1] = buf;
@@ -154,7 +153,6 @@ static int micjack_status_check(void)
 		snd_soc_dapm_enable_pin(&priv->codec->dapm, "DMIC");
 		ret = 0;
 	}
-	snd_soc_dapm_sync(&priv->codec->dapm);
 
 	envp[0] = "NAME=microphone";
 	envp[1] = buf;
-- 
1.7.5.4

